### PR TITLE
[fix](move-memtable) disable move memtable when light schema change is false

### PR DESCRIPTION
**Pull request by sollhui**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by HHoflittlefish777** [comment]
run buildall

**Comment by HHoflittlefish777** [comment]
run buildall

**Review by kaijchen — APPROVED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/load/loadv2/BrokerLoadJob.java [hunk 1]
```diff
@@ -209,13 +209,15 @@ private void createLoadingTask(Database db, BrokerPendingTaskAttachment attachme
                 List<BrokerFileGroup> brokerFileGroups = entry.getValue();
                 long tableId = aggKey.getTableId();
                 OlapTable table = (OlapTable) db.getTableNullable(tableId);
+                boolean isEnableMemtableOnSinkNode = !((OlapTable) table).getTableProperty().getUseSchemaLightChange()
+                        ? false : this.enableMemTableOnSinkNode;
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/load/loadv2/BrokerLoadJob.java:213]:
```suggestion
                boolean isEnableMemtableOnSinkNode = ((OlapTable) table).getTableProperty().getUseSchemaLightChange()
                        ? this.enableMemTableOnSinkNode: false;
```

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/InsertIntoTableCommand.java [hunk 2]
```diff
@@ -145,13 +148,20 @@ private void runInternal(ConnectContext ctx, StmtExecutor executor) throws Excep
...
             LogicalPlanAdapter logicalPlanAdapter = new LogicalPlanAdapter(logicalQuery, ctx.getStatementContext());
             NereidsPlanner planner = new NereidsPlanner(ctx.getStatementContext());
-            planner.plan(logicalPlanAdapter, ctx.getSessionVariable().toThrift());
+            StatementBase parsedStmt = executor.getParsedStmt();
+            TQueryOptions queryOptions = ctx.getSessionVariable().toThrift();
+            boolean isEnableMemtableOnSinkNode =
+                        ! ((OlapTable) ((InsertStmt) parsedStmt).getTargetTable())
+                            .getTableProperty()
+                            .getUseSchemaLightChange()
+                        ? false : queryOptions.isEnableMemtableOnSinkNode();
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/InsertIntoTableCommand.java:157]:
```suggestion
            boolean isEnableMemtableOnSinkNode =
                        ((OlapTable) ((InsertStmt) parsedStmt).getTargetTable())
                            .getTableProperty()
                            .getUseSchemaLightChange()
                        ? queryOptions.isEnableMemtableOnSinkNode() : false;
```

File: fe/fe-core/src/main/java/org/apache/doris/planner/StreamLoadPlanner.java [hunk 3]
```diff
@@ -319,8 +319,10 @@ public TExecPlanFragmentParams plan(TUniqueId loadId, int fragmentInstanceIdInde
         queryOptions.setEnablePipelineEngine(Config.enable_pipeline_load);
         queryOptions.setBeExecVersion(Config.be_exec_version);
         queryOptions.setIsReportSuccess(taskInfo.getEnableProfile());
-        queryOptions.setEnableMemtableOnSinkNode(taskInfo.isMemtableOnSinkNode());
-
+        boolean isEnableMemtableOnSinkNode =
+                ! destTable.getTableProperty().getUseSchemaLightChange()
+                ? false : taskInfo.isMemtableOnSinkNode();
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/planner/StreamLoadPlanner.java:324]:
```suggestion
                destTable.getTableProperty().getUseSchemaLightChange()
                ? taskInfo.isMemtableOnSinkNode() : false;
```

File: fe/fe-core/src/main/java/org/apache/doris/planner/StreamLoadPlanner.java [hunk 4]
```diff
@@ -542,7 +544,10 @@ public TPipelineFragmentParams planForPipeline(TUniqueId loadId, int fragmentIns
         queryOptions.setEnablePipelineEngine(Config.enable_pipeline_load);
         queryOptions.setBeExecVersion(Config.be_exec_version);
         queryOptions.setIsReportSuccess(taskInfo.getEnableProfile());
-        queryOptions.setEnableMemtableOnSinkNode(taskInfo.isMemtableOnSinkNode());
+        boolean isEnableMemtableOnSinkNode =
+                ! destTable.getTableProperty().getUseSchemaLightChange()
+                ? false : taskInfo.isMemtableOnSinkNode();
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/planner/StreamLoadPlanner.java:549]:
```suggestion
                destTable.getTableProperty().getUseSchemaLightChange()
                ? taskInfo.isMemtableOnSinkNode() : false;
```

File: fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java [hunk 5]
```diff
@@ -1226,6 +1227,14 @@ private void analyzeAndGenerateQueryPlan(TQueryOptions tQueryOptions) throws Use
         }
         profile.getSummaryProfile().setQueryAnalysisFinishTime();
         planner = new OriginalPlanner(analyzer);
+        if (parsedStmt instanceof InsertStmt) {
+            boolean isEnableMemtableOnSinkNode =
+                        ! ((OlapTable) ((InsertStmt) parsedStmt).getTargetTable())
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java:1232]: ditto

**Comment by dataroaring** [comment]
run buildall

**Review by morrySnow — DISMISSED** [review]

**Comment by HHoflittlefish777** [comment]
run buildall

**Comment by HHoflittlefish777** [comment]
run buildall

**Comment by HHoflittlefish777** [comment]
run buildall

**Comment by HHoflittlefish777** [comment]
run buildall

**Comment by HHoflittlefish777** [comment]
run buildall

**Comment by HHoflittlefish777** [comment]
run buildall

**Comment by HHoflittlefish777** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**